pulsar-client-api/src/main/java/org/apache/pulsar/client/api/schema/GenericObject.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.client.api.schema;
+
+import org.apache.pulsar.common.schema.SchemaType;
+
+/**
+ * This is an abstraction over the logical value that is store into a Message.
+ * Pulsar decodes the payload of the Message using the Schema that is configured for the topic.
+ */
+public interface GenericObject {
```

Comment:
Is this interface suppose to be used as a schema type for producers or consumers, .e.g. Consumer<GenericObject> ? If so, can you write some tests that has consumers / producers that uses this new type of Schema?

Reply:
Is there a reason why we need this interface as all.  Shouldn't having these additional methods in `GenericRecord` suffice?

Reply:
@jerrypeng 
the idea is to let users user "GenericObject" instead of GenericRecord while dealing with "Any object", on the Consumer/Sink side, because GenericRecord for many users is related to a Struct type (like AVRO GenericRecord), and it carries "getFields()"

`If so, can you write some tests that has consumers / producers that uses this new type of Schema?`
I added tests about GenericObjectWrapper.

The main goal is to have Sink<GenericObject> and having the ability to write Sinks that are not bound to a specific Schema at compile time.